### PR TITLE
OAK-8309: SLOW_QUERY_COUNT don't get updated for each slow query

### PR DESCRIPTION
**Pull request by tihom88**

### CODE REVIEW

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/RuntimeNodeTraversalException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.apache.jackrabbit.oak.query;
+
+public class RuntimeNodeTraversalException extends RuntimeException {
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/query/RuntimeNodeTraversalException.java]: Could you extend UnsupportedOperationException? That would make it compatible with the previously thrown exception, UnsupportedOperationException. (Some code is / was catching UnsupportedOperationException.)

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/SelectorImpl.java [hunk 2]
```diff
@@ -541,6 +558,21 @@ private boolean nextInternal() {
         return false;
     }
 
+    private void totalQueryStats (QueryEngineSettings queryEngineSettings) {
+        if (updateTotalQueryHistogram) {
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/SelectorImpl.java:562]: (here is a space between "totalQueryStats" and "(" which is not necessary)

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/SelectorImpl.java [hunk 3]
```diff
@@ -541,6 +558,21 @@ private boolean nextInternal() {
...
+            updateTotalQueryHistogram = false;
+            HistogramStats histogramStats = queryEngineSettings.getStatisticsProvider().getHistogram(SLOW_QUERY_PERCENTILE_METRICS_NAME, StatsOptions.METRICS_ONLY);
+            histogramStats.update(TOTAL_QUERY_HISTOGRAM);
+        }
+    }
+
+    private void addSlowQueryStats(QueryEngineSettings queryEngineSettings){
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/SelectorImpl.java]: (here is no space between ")" and "{"... By the way a simpley way to ensure consistent formatting is to use the IDE auto-format feature.)

**Review by thomasmueller — COMMENTED** [review]
Other than the inline comments, the patch looks perfect! Thanks a lot!

File: oak-core/src/main/java/org/apache/jackrabbit/oak/query/RuntimeNodeTraversalException.java (hunk 1 above)
Comment (tihom88) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/query/RuntimeNodeTraversalException.java]: Done

**Pull request closed without merging**